File: Java-SpringBoot/BE-02-DB/java-21-springboot-4.0/src/main/java/com/flyrank/taskservice/dto/TaskCreateRequest.java
package com.flyrank.taskservice.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;

@Schema(description = "Payload accepted when creating a new task")
public record TaskCreateRequest(
        @Schema(description = "Task title (required, non-empty)", example = "Build Spring Boot CRUD API", requiredMode = Schema.RequiredMode.REQUIRED)
        @NotBlank(message = "title is required and cannot be empty")
        String title
) {}

File: Java-SpringBoot/BE-02-DB/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/taskservice/dto/TaskCreateRequest.kt
package com.flyrank.taskservice.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank

@Schema(description = "Payload accepted when creating a new task")
data class TaskCreateRequest(
    @field:Schema(description = "Task title (required, non-empty)", example = "Build Spring Boot CRUD API", requiredMode = Schema.RequiredMode.REQUIRED)
    @field:NotBlank(message = "title is required and cannot be empty")
    val title: String? = null
)
